### PR TITLE
[WIP] Refactor Cirrus Daemons / Cleanup

### PR DESCRIPTION
**Pull request by fassadlr**
DONT MERGE until all testing has completed.

### CODE REVIEW

File: src/Stratis.CirrusPegD/Program.cs [hunk 1]
```diff
@@ -113,16 +105,15 @@ private static IFullNode GetSidechainFullNode(string[] args)
...
-            ((PoAConsensusOptions)nodeSettings.Network.Consensus.Options).AutoKickIdleMembers = enableFedKicking;
-
             IFullNode node = new FullNodeBuilder()
                 .UseNodeSettings(nodeSettings)
                 .UseBlockStore()
-                .SetCounterChainNetwork(MainChainNetworks[nodeSettings.Network.NetworkType]())
-                .UseFederatedPegPoAMining()
+                .SetCounterChainNetwork(StraxNetwork.MainChainNetworks[nodeSettings.Network.NetworkType]())
+                .AddPoAFeature()
+                .UsePoAConsensus()
```
Comment (quantumagi) [inline src/Stratis.CirrusPegD/Program.cs:113]: Can these be combined?

File: src/Stratis.SmartContracts.Tests.Common/SmartContractPoARunner.cs [hunk 2]
```diff
@@ -43,8 +45,10 @@ public override void BuildNode()
                             {
                                 options.UseReflectionExecutor();
                             })
-                            .UseSmartContractPoAConsensus()
-                            .UseSmartContractPoAMining(true)
+                            .UsePoAConsensus()
+                            .AddPoAFeature()
```
Comment (quantumagi) [inline src/Stratis.SmartContracts.Tests.Common/SmartContractPoARunner.cs:49]: Non-standard ordering.

**Review by quantumagi — APPROVED** [review]

File: src/Stratis.CirrusPegD/Program.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.CirrusPegD/Program.cs:113]: Good point, I will add this and that other change as well to a following PR.

Comment (fassadlr) [reply src/Stratis.CirrusPegD/Program.cs:113]: Ah I see now that we can't as 2 fluent methods add different features to the node (PoaFeature and ConsenusFeature)

**Pull request merged**